### DISCUSSION TITLE
Disscussion about Proton builds

### DISCUSSION THREAD

**Discussion by Ahmed-E-86** [Kron4ek/Wine-Builds#110] [General]
I am actively working on a personal project, and Proton builds are a vital part of it. I have found all the Proton builds that I need in your repo except 4.11-13 (4.11-12 is the last one in your repo).

I tried to build the other required components like dxvk and vkd3d, but half of them failed to be compiled, and the one I compiled successfully are noticebally different in size than the dll files available in official Proton builds.

One example: `d3d11.dll` (x64) available with official Proton 8.0-3 is 3.6 mb, and d3d11.dll (x64) I compiled on my machine is 3.7 mb!

According to Proton git repo, that is the exact commit for dxvk v2.2-34-g4d254b13-> https://github.com/doitsujin/dxvk/tree/4d254b13bee632e3f5c274e0a7d1aa919238d6a1 and that is exactly what I compiled.

Here is what I did to compile it:
```
git clone --recursive https://github.com/doitsujin/dxvk.git
cd dxvk
git checkout 4d254b13bee632e3f5c274e0a7d1aa919238d6a1
./package-release.sh master . --no-package
```

**Comment by Ahmed-E-86** [comment]
Here is all the Proton build that I want to use with my project:

- proton-3.16-9
- proton-4.11-13
- proton-5.0-10
- proton-5.13-6
- proton-6.3-8
- proton-7.0-1
- proton-7.0-3
- proton-7.0-4
- proton-7.0-5
- proton-7.0-6
- proton-8.0-1
- proton-8.0-2
- proton-8.0-3
- proton-stable (proton-8.0-3)
- proton-experimental

`Proton-experimental` is going to be the most painful, because it gets updated regularly, but right now I am struggling to gather all the components that I need for the regular Proton releases.

**Comment by Kron4ek** [comment]
> d3d11.dll (x64) available with official Proton 8.0-3 is 3.6 mb, and d3d11.dll (x64) I compiled on my machine is 3.7 mb!

Different compiler versions may produce different machine code, which may explain the difference in size.

**Comment by Ahmed-E-86** [comment]
If I build all the Proton builds in the Proton git repo, then I take all native dll files of dxvk, vkd3d, and other components, and use them with your Proton builds, will they work the same as they work with Steam? Or do I need something else?

**Reply by Kron4ek** [reply]
I don't know, i haven't tried to do this.

**Reply by Ahmed-E-86** [reply]
I already tested your Proton builds up to Proton 6.3-8, and everything was working fine except for controller support for some Proton versions, but I do not remember which versions right now. Also, I tested the first version of Proton 7.0, and it was spamming a lot of weird error, that is where I stopped using Proton, and I have been using your wine-tkg builds instead.

Anyway, I will test all your Proton builds at my own pace, and I will tell you the results here in this discussion to get a second opinion if you do not mind.

**Comment by Kron4ek** [comment]
You can use this - https://github.com/Kron4ek/proton-archive/releases

**Reply by Ahmed-E-86** [reply]
OMG!! Thank you so much! You saved me a lot of time. That is incredibly amazing. I owe you a big one.
